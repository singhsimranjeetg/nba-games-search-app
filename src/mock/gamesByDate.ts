//resutls.api.games
export const GamesData = 
  [
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9203",
      "startTimeUTC": "2021-12-16T00:00:00.000Z",
      "endTimeUTC": "2021-12-16T02:23:00.000Z",
      "arena": "Rocket Mortgage FieldHouse",
      "city": "Cleveland",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:12",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "14",
        "shortName": "HOU",
        "fullName": "Houston Rockets",
        "nickName": "Rockets",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/d/de/Houston_Rockets_logo_2003.png/330px-Houston_Rockets_logo_2003.png",
        "score": {
          "points": "89"
        }
      },
      "hTeam": {
        "teamId": "7",
        "shortName": "CLE",
        "fullName": "Cleveland Cavaliers",
        "nickName": "Cavaliers",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/0/06/Cavs_de_Cleveland_logo_2017.png/150px-Cavs_de_Cleveland_logo_2017.png",
        "score": {
          "points": "124"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9204",
      "startTimeUTC": "2021-12-16T00:00:00.000Z",
      "endTimeUTC": "2021-12-16T02:14:00.000Z",
      "arena": "Amway Center",
      "city": "Orlando",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:04",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "1",
        "shortName": "ATL",
        "fullName": "Atlanta Hawks",
        "nickName": "Hawks",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/e/ee/Hawks_2016.png",
        "score": {
          "points": "111"
        }
      },
      "hTeam": {
        "teamId": "26",
        "shortName": "ORL",
        "fullName": "Orlando Magic",
        "nickName": "Magic",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/b/bd/Orlando_Magic_logo_2010.png",
        "score": {
          "points": "99"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9205",
      "startTimeUTC": "2021-12-16T00:00:00.000Z",
      "endTimeUTC": "2021-12-16T02:25:00.000Z",
      "arena": "Wells Fargo Center",
      "city": "Philadelphia",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:14",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "20",
        "shortName": "MIA",
        "fullName": "Miami Heat",
        "nickName": "Heat",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/1/1c/Miami_Heat_-_Logo.svg/1200px-Miami_Heat_-_Logo.svg.png",
        "score": {
          "points": "101"
        }
      },
      "hTeam": {
        "teamId": "27",
        "shortName": "PHI",
        "fullName": "Philadelphia 76ers",
        "nickName": "76ers",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/4/48/76ers_2016.png",
        "score": {
          "points": "96"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9206",
      "startTimeUTC": "2021-12-16T00:30:00.000Z",
      "endTimeUTC": "2021-12-16T03:20:00.000Z",
      "arena": "American Airlines Center",
      "city": "Dallas",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:35",
      "currentPeriod": "5/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "17",
        "shortName": "LAL",
        "fullName": "Los Angeles Lakers",
        "nickName": "Lakers",
        "logo": "https://upload.wikimedia.org/wikipedia/commons/thumb/3/3c/Los_Angeles_Lakers_logo.svg/220px-Los_Angeles_Lakers_logo.svg.png",
        "score": {
          "points": "107"
        }
      },
      "hTeam": {
        "teamId": "8",
        "shortName": "DAL",
        "fullName": "Dallas Mavericks",
        "nickName": "Mavericks",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/b/b8/Mavericks_de_Dallas_logo.svg/150px-Mavericks_de_Dallas_logo.svg.png",
        "score": {
          "points": "104"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9207",
      "startTimeUTC": "2021-12-16T01:00:00.000Z",
      "endTimeUTC": "2021-12-16T03:10:00.000Z",
      "arena": "Fiserv Forum",
      "city": "Milwaukee",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:00",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "15",
        "shortName": "IND",
        "fullName": "Indiana Pacers",
        "nickName": "Pacers",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/c/cf/Pacers_de_l%27Indiana_logo.svg/1180px-Pacers_de_l%27Indiana_logo.svg.png",
        "score": {
          "points": "99"
        }
      },
      "hTeam": {
        "teamId": "21",
        "shortName": "MIL",
        "fullName": "Milwaukee Bucks",
        "nickName": "Bucks",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/3/34/Bucks2015.png",
        "score": {
          "points": "114"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9208",
      "startTimeUTC": "2021-12-16T01:00:00.000Z",
      "endTimeUTC": "2021-12-16T03:37:00.000Z",
      "arena": "Paycom Center",
      "city": "Oklahoma City",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:27",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "23",
        "shortName": "NOP",
        "fullName": "New Orleans Pelicans",
        "nickName": "Pelicans",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/2/21/New_Orleans_Pelicans.png/200px-New_Orleans_Pelicans.png",
        "score": {
          "points": "113"
        }
      },
      "hTeam": {
        "teamId": "25",
        "shortName": "OKC",
        "fullName": "Oklahoma City Thunder",
        "nickName": "Thunder",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/4/4f/Thunder_d%27Oklahoma_City_logo.svg/1200px-Thunder_d%27Oklahoma_City_logo.svg.png",
        "score": {
          "points": "110"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9988",
      "startTimeUTC": "2021-12-16T01:30:00.000Z",
      "endTimeUTC": "2021-12-16T03:46:00.000Z",
      "arena": "AT&T Center",
      "city": "San Antonio",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:06",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "5",
        "shortName": "CHA",
        "fullName": "Charlotte Hornets",
        "nickName": "Hornets",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/f/f3/Hornets_de_Charlotte_logo.svg/1200px-Hornets_de_Charlotte_logo.svg.png",
        "score": {
          "points": "131"
        }
      },
      "hTeam": {
        "teamId": "31",
        "shortName": "SAS",
        "fullName": "San Antonio Spurs",
        "nickName": "Spurs",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/0/0e/San_Antonio_Spurs_2018.png",
        "score": {
          "points": "115"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9210",
      "startTimeUTC": "2021-12-16T02:00:00.000Z",
      "endTimeUTC": "2021-12-16T04:23:00.000Z",
      "arena": "Ball Arena",
      "city": "Denver",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:08",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "22",
        "shortName": "MIN",
        "fullName": "Minnesota Timberwolves",
        "nickName": "Timberwolves",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/d/d9/Timberwolves_du_Minnesota_logo_2017.png/200px-Timberwolves_du_Minnesota_logo_2017.png",
        "score": {
          "points": "124"
        }
      },
      "hTeam": {
        "teamId": "9",
        "shortName": "DEN",
        "fullName": "Denver Nuggets",
        "nickName": "Nuggets",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/3/35/Nuggets_de_Denver_2018.png/180px-Nuggets_de_Denver_2018.png",
        "score": {
          "points": "107"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9211",
      "startTimeUTC": "2021-12-16T03:00:00.000Z",
      "endTimeUTC": "2021-12-16T05:29:00.000Z",
      "arena": "Vivint Arena",
      "city": "Salt Lake City",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:14",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "16",
        "shortName": "LAC",
        "fullName": "LA Clippers",
        "nickName": "Clippers",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/d/d6/Los_Angeles_Clippers_logo_2010.png",
        "score": {
          "points": "103"
        }
      },
      "hTeam": {
        "teamId": "40",
        "shortName": "UTA",
        "fullName": "Utah Jazz",
        "nickName": "Jazz",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/3/3b/Jazz_de_l%27Utah_logo.png",
        "score": {
          "points": "124"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9212",
      "startTimeUTC": "2021-12-16T03:00:00.000Z",
      "endTimeUTC": "2021-12-16T05:32:00.000Z",
      "arena": "Moda Center",
      "city": "Portland",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:21",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "19",
        "shortName": "MEM",
        "fullName": "Memphis Grizzlies",
        "nickName": "Grizzlies",
        "logo": "https://upload.wikimedia.org/wikipedia/en/thumb/f/f1/Memphis_Grizzlies.svg/1200px-Memphis_Grizzlies.svg.png",
        "score": {
          "points": "113"
        }
      },
      "hTeam": {
        "teamId": "29",
        "shortName": "POR",
        "fullName": "Portland Trail Blazers",
        "nickName": "Trail Blazers",
        "logo": "https://upload.wikimedia.org/wikipedia/en/thumb/2/21/Portland_Trail_Blazers_logo.svg/1200px-Portland_Trail_Blazers_logo.svg.png",
        "score": {
          "points": "103"
        }
      }
    },
    {
      "seasonYear": "2021",
      "league": "standard",
      "gameId": "9213",
      "startTimeUTC": "2021-12-16T03:00:00.000Z",
      "endTimeUTC": "2021-12-16T05:28:00.000Z",
      "arena": "Golden 1 Center",
      "city": "Sacramento",
      "country": "USA",
      "clock": "",
      "gameDuration": "2:17",
      "currentPeriod": "4/4",
      "halftime": "0",
      "EndOfPeriod": "0",
      "seasonStage": "2",
      "statusShortGame": "3",
      "statusGame": "Finished",
      "vTeam": {
        "teamId": "41",
        "shortName": "WAS",
        "fullName": "Washington Wizards",
        "nickName": "Wizards",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/archive/d/d6/20161212034849%21Wizards2015.png",
        "score": {
          "points": "105"
        }
      },
      "hTeam": {
        "teamId": "30",
        "shortName": "SAC",
        "fullName": "Sacramento Kings",
        "nickName": "Kings",
        "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/9/95/Kings_de_Sacramento_logo.svg/1200px-Kings_de_Sacramento_logo.svg.png",
        "score": {
          "points": "119"
        }
      }
    }
  ]
