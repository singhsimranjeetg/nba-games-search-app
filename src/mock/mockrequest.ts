import { GamesData } from "./gamesByDate";

export const searchMock = (
  input: RequestInfo,
  init?: RequestInit | undefined
): Promise<Response> =>
  new Promise((resolve) =>
    resolve({
      json: () => Promise.resolve(GamesData),
    } as Response)
  );