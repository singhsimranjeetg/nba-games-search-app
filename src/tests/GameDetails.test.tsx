import React from 'react';
import { shallow } from 'enzyme';
import GameDetails from '../Components/GameDetails.component';
//mock
import { GameData } from '../mock/gamedata.mock';

it('Snapshot tests GameDetails', () => {
  expect(
    shallow(<GameDetails game={GameData.api.game[0]} />)
  ).toMatchSnapshot();
});
