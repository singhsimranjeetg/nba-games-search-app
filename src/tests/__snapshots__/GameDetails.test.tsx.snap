// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests GameDetails 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <Team
        key="2"
        team={
          Object {
            "allStar": "0",
            "fullName": "Boston Celtics",
            "leaders": Array [
              Object {
                "name": "Daniel Theis",
                "playerId": "886",
                "points": "8",
              },
              Object {
                "name": "Daniel Theis",
                "playerId": "886",
                "rebounds": "3",
              },
              Object {
                "assists": "4",
                "name": "Kemba Walker",
                "playerId": "538",
              },
              Object {
                "name": "Jaylen Brown",
                "playerId": "75",
                "rebounds": "10",
              },
              Object {
                "assists": "2",
                "name": "Enes Freedom",
                "playerId": "291",
              },
              Object {
                "name": "Kemba Walker",
                "playerId": "538",
                "points": "39",
              },
              Object {
                "name": "Jayson Tatum",
                "playerId": "882",
                "rebounds": "9",
              },
              Object {
                "assists": "5",
                "name": "Marcus Smart",
                "playerId": "486",
              },
            ],
            "logo": "https://upload.wikimedia.org/wikipedia/fr/thumb/6/65/Celtics_de_Boston_logo.svg/1024px-Celtics_de_Boston_logo.svg.png",
            "nbaFranchise": "1",
            "nickname": "Celtics",
            "score": Object {
              "linescore": Array [
                "30",
                "27",
                "34",
                "30",
              ],
              "loss": "4",
              "points": "121",
              "seriesLoss": "0",
              "seriesWin": "1",
              "win": "13",
            },
            "shortName": "BOS",
            "teamId": "2",
          }
        }
      />
      <span>
        VS
      </span>
      <Team
        key="4"
        team={
          Object {
            "allStar": "0",
            "fullName": "Brooklyn Nets",
            "leaders": Array [
              Object {
                "name": "Jarrett Allen",
                "playerId": "727",
                "points": "7",
              },
              Object {
                "name": "Jarrett Allen",
                "playerId": "727",
                "rebounds": "14",
              },
              Object {
                "assists": "11",
                "name": "Spencer Dinwiddie",
                "playerId": "142",
              },
              Object {
                "name": "Garrett Temple",
                "playerId": "507",
                "points": "22",
              },
              Object {
                "assists": "4",
                "name": "Theo Pinson",
                "playerId": "1013",
              },
            ],
            "logo": "https://upload.wikimedia.org/wikipedia/commons/thumb/4/44/Brooklyn_Nets_newlogo.svg/130px-Brooklyn_Nets_newlogo.svg.png",
            "nbaFranchise": "1",
            "nickname": "Nets",
            "score": Object {
              "linescore": Array [
                "23",
                "40",
                "23",
                "24",
              ],
              "loss": "9",
              "points": "110",
              "seriesLoss": "1",
              "seriesWin": "0",
              "win": "9",
            },
            "shortName": "BKN",
            "teamId": "4",
          }
        }
      />
    </styled.div>
    <div
      className="mx-auto"
    >
      <p
        className="date"
      >
        2019-11-28
      </p>
      <p
        className="venue"
      >
        TD Garden
        , 
        Boston
        , 
        USA
      </p>
    </div>
  </styled.div>
</Fragment>
`;
